Makefile

```diff
@@ -8,7 +8,7 @@ export TERRAFORM_VERSION := 1.5.5
 
 export TERRAFORM_PROVIDER_SOURCE ?= hashicorp/vault
 export TERRAFORM_PROVIDER_REPO ?= https://github.com/hashicorp/terraform-provider-vault
-export TERRAFORM_PROVIDER_VERSION ?= 3.25.0
+export TERRAFORM_PROVIDER_VERSION ?= 4.2.0
```

Comment:
It looks like there's already been a discussion about this part. I want to mention a few things. During such major version bumps, it would be nice if the resources with changes in the API were tested and there was a note in the PR description that these were tested.

Also, as already mentioned, it would be beneficial for users to clearly state the relevant breaking changes in the Release Notes.

As a final point of discussion, since this upgrade causes breaking change, does it make sense to do a major version bump after the relevant PR merge? So, releasing `1.0.0`.

Reply:
> It looks like there's already been a discussion about this part. I want to mention a few things. During such major version bumps, it would be nice if the resources with changes in the API were tested and there was a note in the PR description that these were tested.
> 
> Also, as already mentioned, it would be beneficial for users to clearly state the relevant breaking changes in the Release Notes.
> 
> As a final point of discussion, since this upgrade causes breaking change, does it make sense to do a major version bump after the relevant PR merge? So, releasing `1.0.0`.

Thanks for the review @sergenyalcin , and it makes sense to me to do a major version bump 👍 I will continue with testing the resources with changes to the API and update the PR description ASAP.

Reply:
@sergenyalcin Update: I noticed a couple issues with resources going out of sync while testing with the new terraform (v4.5.0). In the interest of time and priority, I reverted the changes for the terraform provider version upgrade and instead would like this PR to focus on solely addressing #12.